--- /net/minecraft/stats
+++ /net/minecraft/stats
@@ -48,6 +48,13 @@
     }
 
     public void unlockAchievement(EntityPlayer var1, StatBase var2, int var3) {
+        if (var2.isAchievement()) {
+            xyz.openmodloader.event.impl.player.EventAchievement event = xyz.openmodloader.event.impl.player.EventAchievement.handle(var1, (Achievement) var2, var3);
+            var2 = event.achievement;
+            var3 = event.data;
+            if (event.isCanceled()) return;
+        }
+
         TupleIntJsonSerializable var4 = (TupleIntJsonSerializable)this.statsData.get(var2);
         if(var4 == null) {
             var4 = new TupleIntJsonSerializable();
