--- /net/minecraft/stats
+++ /net/minecraft/stats
@@ -67,8 +67,15 @@
     }
 
     public void unlockAchievement(EntityPlayer var1, StatBase var2, int var3) {
-        int var4 = var2.isAchievement()?this.readStat(var2):0;
+        if (var2.isAchievement()) {
+            xyz.openmodloader.event.impl.player.EventAchievement event = xyz.openmodloader.event.impl.player.EventAchievement.handle(var1, (Achievement) var2, var3);
+            var2 = event.achievement;
+            var3 = event.data;
+            if (event.isCanceled()) return;
+        }
+
         super.unlockAchievement(var1, var2, var3);
+        int var4 = var2.isAchievement()?this.readStat(var2):0;
         this.dirty.add(var2);
         if(var2.isAchievement() && var4 == 0 && var3 > 0) {
             this.hasUnsentAchievement = true;
