--- /net/minecraft/server/dedicated
+++ /net/minecraft/server/dedicated
@@ -85,6 +85,7 @@
     }
 
     public boolean startServer() throws IOException {
+        xyz.openmodloader.OpenModLoader.minecraftConstruction(new xyz.openmodloader.server.OMLServerHandler(this));
         Thread var1 = new Thread("Server console handler") {
             public void run() {
                 BufferedReader var1 = new BufferedReader(new InputStreamReader(System.in));
@@ -103,6 +104,7 @@
         var1.setDaemon(true);
         var1.start();
         LOGGER.info("Starting minecraft server version 1.10.2");
+        xyz.openmodloader.OpenModLoader.getEventBus().post(new xyz.openmodloader.event.impl.ServerEvent.Starting(this));
         if(Runtime.getRuntime().maxMemory() / 1024L / 1024L < 512L) {
             LOGGER.warn("To start the server with more ram, launch it as \"java -Xmx1024M -Xms1024M -jar minecraft_server.jar\"");
         }
@@ -238,6 +240,7 @@
                     var15.start();
                 }
 
+                xyz.openmodloader.OpenModLoader.getEventBus().post(new xyz.openmodloader.event.impl.ServerEvent.Started(this));
                 return true;
             }
         }
