--- /net/minecraft/world/chunk/storage
+++ /net/minecraft/world/chunk/storage
@@ -279,6 +279,7 @@
             var3.setTag("TileTicks", var28);
         }
 
+        xyz.openmodloader.world.storage.OMLSaveHandler.writeChunkToNBT(var1, var3);
     }
 
     private Chunk readChunkFromNBT(World var1, NBTTagCompound var2) {
@@ -353,6 +354,7 @@
             }
         }
 
+        xyz.openmodloader.world.storage.OMLSaveHandler.readChunkFromNBT(var5, var2);
         return var5;
     }
 
