--- /net/minecraft/entity/passive
+++ /net/minecraft/entity/passive
@@ -77,6 +77,7 @@
     private static final DataParameter<Integer> HORSE_VARIANT = EntityDataManager.<Integer>createKey(EntityHorse.class, DataSerializers.VARINT);
     private static final DataParameter<Optional<UUID>> OWNER_UNIQUE_ID = EntityDataManager.<Optional<UUID>>createKey(EntityHorse.class, DataSerializers.OPTIONAL_UNIQUE_ID);
     private static final DataParameter<Integer> HORSE_ARMOR = EntityDataManager.<Integer>createKey(EntityHorse.class, DataSerializers.VARINT);
+    private static final DataParameter<Optional<ItemStack>> ARMOR_STACK = EntityDataManager.<Optional<ItemStack>>createKey(EntityHorse.class, DataSerializers.OPTIONAL_ITEM_STACK);
     private static final String[] HORSE_TEXTURES = new String[]{"textures/entity/horse/horse_white.png", "textures/entity/horse/horse_creamy.png", "textures/entity/horse/horse_chestnut.png", "textures/entity/horse/horse_brown.png", "textures/entity/horse/horse_black.png", "textures/entity/horse/horse_gray.png", "textures/entity/horse/horse_darkbrown.png"};
     private static final String[] HORSE_TEXTURES_ABBR = new String[]{"hwh", "hcr", "hch", "hbr", "hbl", "hgr", "hdb"};
     private static final String[] HORSE_MARKING_TEXTURES = new String[]{null, "textures/entity/horse/horse_markings_white.png", "textures/entity/horse/horse_markings_whitefield.png", "textures/entity/horse/horse_markings_whitedots.png", "textures/entity/horse/horse_markings_blackdots.png"};
@@ -136,6 +137,7 @@
         this.dataManager.register(HORSE_VARIANT, Integer.valueOf(0));
         this.dataManager.register(OWNER_UNIQUE_ID, Optional.absent());
         this.dataManager.register(HORSE_ARMOR, Integer.valueOf(HorseArmorType.NONE.getOrdinal()));
+        this.dataManager.register(ARMOR_STACK, Optional.absent());
     }
 
     public void setType(HorseType var1) {
@@ -258,6 +260,7 @@
     public void setHorseArmorStack(ItemStack var1) {
         HorseArmorType var2 = HorseArmorType.getByItemStack(var1);
         this.dataManager.set(HORSE_ARMOR, Integer.valueOf(var2.getOrdinal()));
+        this.dataManager.set(ARMOR_STACK, Optional.fromNullable(var1));
         this.resetTexturePrefix();
         if(!this.worldObj.isRemote) {
             this.getEntityAttribute(SharedMonsterAttributes.ARMOR).removeModifier(ARMOR_MODIFIER_UUID);
@@ -567,7 +570,8 @@
         }
 
         HorseArmorType var5 = this.getHorseArmorType();
-        this.horseTexturesArray[2] = var5.getTextureName();
+        final ItemStack armorStack = this.dataManager.get(ARMOR_STACK).orNull();
+        this.horseTexturesArray[2] = armorStack != null && armorStack.getItem() instanceof xyz.openmodloader.item.HorseArmor ? ((xyz.openmodloader.item.HorseArmor) armorStack.getItem()).getArmorTexture(this, armorStack) :var5.getTextureName();
         this.texturePrefix = this.texturePrefix + var5.getHash();
         this.hasTexture = true;
     }
