--- /net/minecraft/entity/passive
+++ /net/minecraft/entity/passive
@@ -7,11 +7,11 @@
 import xyz.openmodloader.launcher.strippable.Side;
 import xyz.openmodloader.launcher.strippable.Strippable;
 
-public enum HorseArmorType {
-    NONE(0),
-    IRON(5, "iron", "meo"),
-    GOLD(7, "gold", "goo"),
-    DIAMOND(11, "diamond", "dio");
+public class HorseArmorType {
+    public static final HorseArmorType NONE = new HorseArmorType(0);
+    public static final HorseArmorType IRON = new HorseArmorType(5, "iron", "meo");
+    public static final HorseArmorType GOLD = new HorseArmorType(7, "gold", "goo");
+    public static final HorseArmorType DIAMOND = new HorseArmorType(11, "diamond", "dio");
 
     private final String textureName;
     private final String hash;
@@ -23,14 +23,14 @@
         this.hash = "";
     }
 
-    private HorseArmorType(int var3, String var4, String var5) {
+    public HorseArmorType(int var3, String var4, String var5) {
         this.protection = var3;
         this.textureName = "textures/entity/horse/armor/horse_armor_" + var4 + ".png";
         this.hash = var5;
     }
 
     public int getOrdinal() {
-        return this.ordinal();
+        return xyz.openmodloader.registry.Registries.get(HorseArmorType.class).getIDForObject(this);
     }
 
     @Strippable(
@@ -53,18 +53,18 @@
     }
 
     public static HorseArmorType getByOrdinal(int var0) {
-        return values()[var0];
+        return xyz.openmodloader.registry.Registries.get(HorseArmorType.class).getObjectById(var0);
     }
 
     public static HorseArmorType getByItemStack(@Nullable ItemStack var0) {
-        return var0 == null?NONE:getByItem(var0.getItem());
+        return var0 == null?NONE:var0.getItem() instanceof xyz.openmodloader.item.HorseArmor?((xyz.openmodloader.item.HorseArmor) var0.getItem()).getArmorType(var0):getByItem(var0.getItem());
     }
 
     public static HorseArmorType getByItem(@Nullable Item var0) {
-        return var0 == Items.IRON_HORSE_ARMOR?IRON:(var0 == Items.GOLDEN_HORSE_ARMOR?GOLD:(var0 == Items.DIAMOND_HORSE_ARMOR?DIAMOND:NONE));
+        return var0 instanceof xyz.openmodloader.item.HorseArmor?((xyz.openmodloader.item.HorseArmor) var0).getArmorType(new ItemStack(var0)):var0 == Items.IRON_HORSE_ARMOR?IRON:(var0 == Items.GOLDEN_HORSE_ARMOR?GOLD:(var0 == Items.DIAMOND_HORSE_ARMOR?DIAMOND:NONE));
     }
 
     public static boolean isHorseArmor(@Nullable Item var0) {
-        return getByItem(var0) != NONE;
+        return var0 instanceof xyz.openmodloader.item.HorseArmor || getByItem(var0) != NONE;
     }
 }
