--- /net/minecraft/entity
+++ /net/minecraft/entity
@@ -492,6 +492,7 @@
     }
 
     protected void updateEquipmentIfNeeded(EntityItem var1) {
+        if (!xyz.openmodloader.OpenModLoader.getEventBus().post(new xyz.openmodloader.event.impl.EntityEvent.ItemPickup(this, var1))) return;
         ItemStack var2 = var1.getEntityItem();
         EntityEquipmentSlot var3 = getSlotForItemStack(var2);
         boolean var4 = true;
@@ -582,6 +583,7 @@
         } else {
             EntityPlayer var1 = this.worldObj.getClosestPlayerToEntity(this, -1.0D);
             if(var1 != null) {
+                if (xyz.openmodloader.event.impl.LivingEvent.Despawn.handle(this).isCanceled()) return;
                 double var2 = var1.posX - this.posX;
                 double var4 = var1.posY - this.posY;
                 double var6 = var1.posZ - this.posZ;
@@ -736,6 +738,9 @@
     }
 
     public void setItemStackToSlot(EntityEquipmentSlot var1, @Nullable ItemStack var2) {
+        xyz.openmodloader.event.impl.EquipmentEvent event = xyz.openmodloader.event.impl.EquipmentEvent.Equip.handle(this, var2, var1);
+        if (event == null) return;
+        var1 = event.getSlot(); var2 = event.getEquipment();
         switch(var1.getSlotType()) {
         case HAND:
             this.inventoryHands[var1.getIndex()] = var2;
