--- /net/minecraft/inventory
+++ /net/minecraft/inventory
@@ -356,4 +356,19 @@
 
         this.updateRepairOutput();
     }
+
+    @Override
+    public ItemStack slotClick(int slotNumber, int drag, ClickType clickType, EntityPlayer player) {
+        if (slotNumber == 2) {
+            xyz.openmodloader.event.impl.player.EventTakeSlot.Repair event = xyz.openmodloader.event.impl.player.EventTakeSlot.Repair.handle(player, (ContainerRepair) this);
+            if (this.getSlot(2).canTakeStack(player) && !event.isCanceled()) {
+                this.maximumCost = event.cost;
+                return super.slotClick(slotNumber, drag, clickType, player);
+            } else {
+                return null;
+            }
+        } else {
+            return super.slotClick(slotNumber, drag, clickType, player);
+        }
+    }
 }
