--- /net/minecraft/inventory
+++ /net/minecraft/inventory
@@ -51,6 +51,13 @@
                     }
                 }
 
+                public void putStack(@Nullable ItemStack stack) {
+                    xyz.openmodloader.event.impl.EquipmentEvent event = xyz.openmodloader.event.impl.EquipmentEvent.Equip.handle(thePlayer, stack, var9);
+                    if (event == null) return;
+                    stack = event.getEquipment();
+                    super.putStack(stack);
+                }
+
                 @Nullable
                 @Strippable(
                     side = Side.CLIENT
@@ -131,11 +138,17 @@
                     return null;
                 }
             } else if(var6.getSlotType() == EntityEquipmentSlot.a.ARMOR && !((Slot)this.inventorySlots.get(8 - var6.getIndex())).getHasStack()) {
+                xyz.openmodloader.event.impl.EquipmentEvent event = xyz.openmodloader.event.impl.EquipmentEvent.Equip.handle(thePlayer, var5, var6);
+                if (event == null) return null;
+                var6 = event.getSlot(); var5 = event.getEquipment();
                 int var7 = 8 - var6.getIndex();
                 if(!this.mergeItemStack(var5, var7, var7 + 1, false)) {
                     return null;
                 }
             } else if(var6 == EntityEquipmentSlot.OFFHAND && !((Slot)this.inventorySlots.get(45)).getHasStack()) {
+                xyz.openmodloader.event.impl.EquipmentEvent event = xyz.openmodloader.event.impl.EquipmentEvent.Equip.handle(thePlayer, var5, var6);
+                if (event == null) return null;
+                var6 = event.getSlot(); var5 = event.getEquipment();
                 if(!this.mergeItemStack(var5, 45, 46, false)) {
                     return null;
                 }
