--- /net/minecraft/entity/item
+++ /net/minecraft/entity/item
@@ -130,6 +130,9 @@
     }
 
     public void setItemStackToSlot(EntityEquipmentSlot var1, @Nullable ItemStack var2) {
+        xyz.openmodloader.event.impl.EquipmentEvent event = xyz.openmodloader.event.impl.EquipmentEvent.Equip.handle(this, var2, var1);
+        if (event == null) return;
+        var1 = event.getSlot(); var2 = event.getEquipment();
         switch(var1.getSlotType()) {
         case HAND:
             this.playEquipSound(var2);
