--- /net/minecraft/item
+++ /net/minecraft/item
@@ -218,7 +218,7 @@
             return false;
         } else {
             if(var1 > 0) {
-                int var3 = EnchantmentHelper.getEnchantmentLevel(Enchantments.UNBREAKING, this);
+                int var3 = EnchantmentHelper.getEnchantmentLevel(Enchantments.UNBREAKING, this, null);
                 int var4 = 0;
 
                 for(int var5 = 0; var3 > 0 && var5 < var1; ++var5) {
@@ -773,4 +773,7 @@
             return false;
         }
     }
+
+    public boolean equals(Object obj) { return this == obj || obj instanceof ItemStack && isItemStackEqual((ItemStack) obj); }
+    public int hashCode() { return java.util.Objects.hash(Item.getIdFromItem(item), stackSize, itemDamage, stackTagCompound); }
 }
