--- /net/minecraft/item
+++ /net/minecraft/item
@@ -137,8 +137,8 @@
 import xyz.openmodloader.launcher.strippable.Strippable;
 
 public class Item {
-    public static final RegistryNamespaced<ResourceLocation, Item> REGISTRY = new RegistryNamespaced();
-    private static final Map<Block, Item> BLOCK_TO_ITEM = Maps.newHashMap();
+    public static final xyz.openmodloader.registry.AutomaticNamespacedRegistry<ResourceLocation, Item> REGISTRY = xyz.openmodloader.registry.Registries.register(Item.class);
+    private static final Map<Block, Item> BLOCK_TO_ITEM = xyz.openmodloader.registry.Registries.getBlockItemMap();
     private static final IItemPropertyGetter DAMAGED_GETTER = new IItemPropertyGetter() {
         @Strippable(
             side = Side.CLIENT
@@ -463,6 +463,14 @@
         return HashMultimap.create();
     }
 
+    // OML
+    /**
+     * Called to tick the armor item when it is being worn.
+     * @param player The player wearing the armor
+     * @param stack The armor stack being worn
+     */
+    public void onArmorTick(EntityPlayer player, ItemStack stack) {}
+
     public static void registerItems() {
         registerItemBlock(Blocks.STONE, (new ItemMultiTexture(Blocks.STONE, Blocks.STONE, new Function<ItemStack, String>() {
             @Nullable
